ecosystem/sep-0013.md

```diff
@@ -0,0 +1,202 @@
+    
+The process is described as a list of actions Robert and Sarah must take. Often
+times, these action *may* be aggregated into a single transaction in
+implementation, but aggregation *must* not change the behavior of the
+un-aggregated version.
+
+### Case 1: Create Account
+In this case, Sarah creates an account for Robert which has 10 FunBucks in it.
+
+1. Robert has a specified set of signers, {R}. For simplicity, assume all of {R}
+   are EDDSA25519 keys.
+1. Sarah wants to pay Robert 10 FunBucks.
+1. Sarah does a transaction which:
+    - creates an deterministic account C with sufficient reserve for the signers {R} and one trust-line.
+    - Adds a trustline for 10 FunBucks to account C
+    - If Auth is required, Sarah includes an AllowTrust C (Sarah must also get FunBucks Issuer to sign this transaction)
+    - Sends 10 FunBucks to C.
+    - Adds the signers {R} specified by Robert.
+1. Robert queries horizon for accounts with the signers {R}, and notices C.
+
+
+ Robert then does whatever Robert wants with the account. Notably, Sarah has
+ paid Robert the required reserve for {R} and a trustline. Perhaps Sarah deducts
+ this from the amount of FunBucks paid, from a future transaction, or Robert
+ pays Sarah back at some point in the future if reimbursement  is required.
+
+### Case 2: No Create Account
+
+In this case, Sarah does not wish to pay reserve for Robert, but wants to
+transfer custody of 10 FunBucks to Robert (and be guaranteed to have her reserve
+returned).
+
+1. Robert has a specified set of signers, {R}. For simplicity, assume all of {R}
+   are EDDSA25519 keys. Robert also specifies an account ID Ra which may or may
+   not exist.
+1. Sarah wants to pay Robert 10 FunBucks.
+1. Sarah checks to see if Ra exists and has a trustline which can receive 10
+   FunBucks.
+1. If such an account exists, Sarah transfers 10 FunBucks to Robert and
+   finishes.
+1. If such an account did not exist, Sarah continues.
+1. Sarah does a transaction with memo Ra which:
+    - creates an deterministic account C with sufficient reserve for 2 signers and one trust-line.
+    - Adds a trustline for 10 FunBucks to account C
+    - If Auth is required, Sarah includes an AllowTrust C (Sarah must also get FunBucks Issuer to sign this transaction)
+    - Sends 10 FunBucks to C.
+    - Adds a preauth at sequence 1 that
+        - Adds Sarah's key to the account
+    - Adds a preauth at sequence 0 that
+        - Optionally, pays transaction fees sufficient for the setup from Ra to C
+        - Sends 10 FunBucks from C to Ra
+        - BumpSequence 0 Ra (to require Robert's signature)
+        - Adds Sarah's key to the account
+        - Removes the preauth at sequence 1
+1. Robert queries horizon for recent transactions with memo Ra, and notices
+   account C.
+1. Robert derives the preauth transaction (it is deterministic).
+1. Robert ensures that Ra is created and has a trustline that can support
```

Comment:
I think it depends on what you mean by "transferred to Ra".

You have an impossible case. Sarah is both unwilling to pay for anything relating to Robert, yet you want Sarah to pay for the account and trustline. Can't be done. Somewhere someone has to pay for something, and you can't have someone pay and not pay at the same time. There are also cases where an account ID specified (post CAP-0007, which will happen anyways) can never exist again.

In the scenario I've given, Sarah guarantees that the transfer to Ra goes through if Robert creates the account with the correct balance. Software can prevent this case from being erroneously triggered by user error.

Legally speaking, the funds are transferred to Robert and Sarah no longer has liability for them. This is a successful deposit.

If you meant something else by deposit, then you should specify what deposit means concretely. A definition this would not satisfy is that the asset is immediately spendable by the recipient, but I've yet to see a CAP-0013 type proposal that satisfies that requirement either. If you don't have "extra" xlm in your account to pay a next TX fee, it turns out you can't transact either. So what does it mean for funds to be "deposited"? Is it merely a UX layer thing?

Largely speaking I'm also 100% skeptical of the 'suggestion' (I'm relabeling the things your calling requirements as suggestions because they're not somehow community-ratified) that funds wind up in exactly the account id specified by Robert. Robert should not care what account ID they are in as long as Robert owns it.  There are a million ways this can work, ranging from Horizon changes to support spend-ability monitoring, Sarah sending Robert a receipt for the account, etc. It's very flexible to meeting the exact need of the user. The suggestions for how it should work are simultaneously over-strict where it doesn't matter and under-strict where it does.

The bar in order for me to be convinced that something like CAP-0013 is a better suited strategy minimally includes a corresponding SEP that gives, in similar level of detail to SEP-0013, definitions for exactly how CAP-0013 could be used for a wallet implementer. This isn't a high bar given that this is already what SEP-0013 is.

I would also like to see the suggestions document you linked refined. I would like to see twice as many suggestions on what properties are necessary (there are more interesting properties that aren't covered) and also a justification for why each suggestion must be met and alternatives will not work practically.

Reply:
> You have an impossible case. Sarah is both unwilling to pay for anything relating to Robert, yet you want Sarah to pay for the account and trustline. Can't be done. Somewhere someone has to pay for something, and you can't have someone pay and not pay at the same time. There are also cases where an account ID specified (post CAP-0007, which will happen anyways) can never exist again.

I don't think Sarah is unwilling to pay for anything related to Robert. I think the whole point is that she is willing to pay for Robert's trustline so the payment can complete in one shot.

Can you respond directly to the comment I made above? The rest of your response is about the document  I linked, and I appreciate the feedback. To get back to my original comment, I'm still not seeing how the flow completes if Robert has no access to XLM from some other source.